app/scripts/modules/core/src/presentation/ReactModal.tsx

```diff
@@ -2,14 +2,7 @@ import React from 'react';
 import { Modal, ModalProps } from 'react-bootstrap';
 import ReactDOM from 'react-dom';
 
-/** The Modal content Component will be passed these two props */
-export interface IModalComponentProps {
-  // Close modal with a result value (i.e., OK button)
-  closeModal?(result?: any): void;
-
-  // Dismiss/reject modal (i.e., Cancel button)
-  dismissModal?(rejectReason?: any): void;
-}
+import { IModalComponentProps } from './modal';
```

Comment:
I think that besides optionally parameterizing the result types (which is done in this PR) the shape of `IModalComponentProps` is remains very useful and has no reason to change. Since it's already got the right shape and name, there's no reason to duplicate it, instead let's just move it to a new home and continue using it.